cpapp_backend/cpapp/services/agent.py

```
@@ -3668,6 +3894,317 @@ def _handle_address_details_completion(self, session_id: str, message: str) -> s
+                    # Find the last AI message
+                    for message_entry in reversed(session_history):
+                        if message_entry.get("type") == "AIMessage":
+                            previous_ai_message = message_entry.get("content", "")
+                            logger.info(f"Session {session_id}: Returning previous AI message from database")
+                            return previous_ai_message
+                
+                # If no previous message found, return a default message
+                logger.warning(f"Session {session_id}: No previous AI message found")
+                return "Please complete the EMI auto-pay setup process."
+                
+        except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)